### PR TITLE
Main.yml respect both web and task deployments

### PR DESCRIPTION
**Pull request by djyasin**
##### SUMMARY

Making rols/installer/main.yml respect both web and task deployment

##### ISSUE TYPE

 - New or Enhanced Feature

### CODE REVIEW

**Review by TheRealHaoLiu — COMMENTED** [review]
lgtm

File: roles/installer/tasks/main.yml [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
   k8s_info:
     api_version: v1
     kind: Deployment
-    name: "{{ ansible_operator_meta.name }}"
+    name: "{{ item }}"
     namespace: "{{ ansible_operator_meta.namespace }}"
+  loop: 
+     - "{{ ansible_operator_meta.namespace }}-web"
+     - "{{ ansible_operator_meta.namespace }}-task"
   register: tower_deployment
```
Comment (TheRealHaoLiu) [inline roles/installer/tasks/main.yml]: this doesnt work here u will just be overriding "tower_deployment"

Comment (TheRealHaoLiu) [reply roles/installer/tasks/main.yml]: try to see what "tower_deployment" is being used for now and how it maps to "web" and "task" deployments

Comment (djyasin) [reply roles/installer/tasks/main.yml]: Thank you! I was concerned about that. I will go investigate "tower_deployment."

File: roles/installer/tasks/main.yml [hunk 2]
```diff
@@ -1,11 +1,10 @@
 ---
-- name: Check for presence of Deployment
+- name: Check for presence of web deployment
   k8s_info:
     api_version: v1
     kind: Deployment
-    name: "{{ ansible_operator_meta.name }}"
+    name: "{{ ansible_operator_meta.name-web }}"
```
Comment (TheRealHaoLiu) [inline roles/installer/tasks/main.yml]: variable ansible_operator_meta.name-web is not a exist 

ansible_operator_meta.name exist 

so this line should be `name: "{{ ansible_operator_meta.name }}-web"`

it appends -web to whatever `ansible_operator_meta.name` resolve to

File: roles/installer/tasks/main.yml [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
     namespace: "{{ ansible_operator_meta.namespace }}"
-  register: tower_deployment
```
Comment (TheRealHaoLiu) [inline roles/installer/tasks/main.yml:8]: this register should still exist or else the following code wont have the `tower_deployment` variable

**Comment by TheRealHaoLiu** [comment]
`Start installation task` should only run under the following condition

- if web and task deployment BOTH **does not exist** run install.yml
- if EITHER web or task deployment exist and auto_upgrade is set to true run install.yml

**Review by thedoubl3j — APPROVED** [review]

**Review by rooftopcellist — APPROVED** [review]

**Comment by TheRealHaoLiu** [comment]
superseded by https://github.com/ansible/awx-operator/pull/1241

**Pull request closed without merging**